pylint/checkers/typecheck.py

```diff
@@ -1559,25 +1559,28 @@ def _check_not_callable(
         Only functions, generators and objects defining __call__ are "callable"
         We ignore instances of descriptors since astroid cannot properly handle them yet
         """
-        if inferred_call and not inferred_call.callable():
-            if isinstance(inferred_call, astroid.Instance):
-                # Don't emit if we can't make sure this object is callable.
-                if not has_known_bases(inferred_call):
-                    return
-
-                if inferred_call.parent and isinstance(
-                    inferred_call.scope(), nodes.ClassDef
-                ):
-                    # Ignore descriptor instances
-                    if "__get__" in inferred_call.locals:
-                        return
-                    # NamedTuple instances are callable
-                    if inferred_call.qname() == "typing.NamedTuple":
-                        return
-
-            self.add_message("not-callable", node=node, args=node.func.as_string())
-        else:
+        # Handle uninferrable calls
+        if not inferred_call or inferred_call.callable():
             self._check_uninferable_call(node)
+            return
+
+        # Ignore non instances
+        if not isinstance(inferred_call, astroid.Instance):
```

Comment:
Sorry I was wrong in my first proposition I edited it. I think there's a guaranteed message for non instances. (The add_message is in the same scope)

Reply:
This doesn't fail any tests.. The coveralls comments seems broken, but are we actually testing this?

Reply:
Relevant commit:
https://github.com/PyCQA/pylint/commit/ea14cad3187aa2efdd0ae7b520b2ba44797c0863

Seems like we can remove it? All `inferred_call`'s that we test are `Instance`'s

Reply:
I think you're right, but it's really hard to reconstruct code example that would trigger a particular piece of code though. (Or "prove" that it's impossible). I hope primer++ will fix this.